src_c/base.c

```diff
@@ -173,43 +173,32 @@ pg_SetDefaultWindowSurface(pgSurfaceObject *);
 static char *
 pg_EnvShouldBlendAlphaSDL2(void);
 
+/* compare compiled to linked, raise python error on incompatibility */
 static int
-pg_CheckSDLVersions(void) /*compare compiled to linked*/
+pg_CheckSDLVersions(void)
 {
     SDL_version compiled;
     SDL_version linked;
 
     SDL_VERSION(&compiled);
     SDL_GetVersion(&linked);
 
-    /*only check the major and minor version numbers.
-      we will relax any differences in 'patch' version.*/
-
-    if (compiled.major != linked.major || compiled.minor != linked.minor) {
+    /* only check the major version, in general major version is bumped for ABI
+     * incompatible changes */
+    if (compiled.major != linked.major) {
         PyErr_Format(PyExc_RuntimeError,
-                     "SDL compiled with version %d.%d.%d, linked to %d.%d.%d",
+                     "ABI incompatibility detected! SDL compiled with "
+                     "%d.%d.%d, linked to %d.%d.%d (major versions should "
+                     "have matched)",
                      compiled.major, compiled.minor, compiled.patch,
                      linked.major, linked.minor, linked.patch);
         return 0;
     }
-    else if (linked.major == 2 && linked.minor == 0 && linked.patch < 14 &&
-             compiled.patch >=
-                 14) {  // major and minor versions match, check edge case
-        /* SDL 2.0.14 replaces some macros with symbols, see
-         * https://github.com/libsdl-org/SDL/commit/316ff3847b4d9d87d9b0aab15321461db0e8ae0b
-         */
-        PyErr_Format(PyExc_RuntimeError,
-                     "Known SDL incompatibility detected! (compiled with "
-                     "version %d.%d.%d, linked to %d.%d.%d)",
-                     compiled.major, compiled.minor, compiled.patch,
-                     linked.major, linked.minor, linked.patch);
-        return 0;
-    }
```

Comment:
Is this no longer the case? 

I had a skim through the linked SDL discussion. Have they re-released SDL 2.0.14 as  SDL 2.14.0 now?
It is a bit confusing.

Reply:
They have not re-released older versions, they only adopted a new versioning scheme starting from 2.24.0 (which is the stable version released after 2.0.22)

Because the versioning system changed like this, this check is is not robust enough anymore (for instance, this case would execute if we have a pygame build compiled with a hypothetical SDL 2.24.14 which linked with 2.0.12, which should still error, but it is not the "known SDL incompatibility" this message tells)

I just completely removed this block because it was merely a special case of the below generic error reporting block, and this did not specify more helpful information anyways. (in both cases, a fresh compile/install would be needed)

Reply:
2.0.14 is still 2.0.14, nothing has changed for past versions (we don't have a time machine, we couldn't change past versions even if we wanted to).

The block that special-cased 2.0.14 (if compiled version >= 2.0.14 and 2.0.14 > linked version then error) looks redundant with the block below it (if compiled version > linked version then error), so it seems valid to remove.

Reply:
It was deliberately special-cased because 1. the original PR only had this case and not the general case and 2. we wanted a special error MSG for this case